src/DateTimeValidator.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+namespace ipl\Validator;
+
+use DateTime;
+use ipl\I18n\Translation;
+
+/**
+ * Validator for date-and-time input controls
+ */
+class DateTimeValidator extends BaseValidator
+{
+    use Translation;
+
+    protected $local;
+
+    /**
+     * Create a new date-and-time input control validator
+     *
+     * @param bool $local
+     */
+    public function __construct($local = true)
+    {
+        $this->local = (bool) $local;
+    }
+
+    /**
+     * Check whether the given date time is valid
+     *
+     * @param   string|DateTime $value
+     *
+     * @return  bool
+     */
+    public function isValid($value)
+    {
+        // Multiple isValid() calls must not stack validation messages
+        $this->clearMessages();
+
+        if (! $value instanceof DateTime && ! is_string($value)) {
+            $this->addMessage($this->translate('Invalid date/time given.'));
+
+            return false;
+        }
+
+        if (! $value instanceof DateTime) {
+            $format = $baseFormat = $this->local === true ? 'Y-m-d\TH:i:s' : DateTime::RFC3339;
```

Comment:
Remove `$baseFormat`, it's redundant.

And maybe make `'Y-m-d\TH:i:s'` a constant. That way it could be changed by a subclass easily and also be referenced outside for further usage.